modcp.php

```diff
@@ -3569,7 +3569,77 @@
 
 			if($post_ip_sql)
 			{
-				$query = $db->simple_select('posts', 'COUNT(pid) AS count', "$post_ip_sql AND visible >= -1");
+				$where_sql = '';
+
+				$unviewable_forums = get_unviewable_forums(true);
+
+				if($unviewable_forums)
+				{
+					$where_sql .= " AND p.fid NOT IN ({$unviewable_forums})";
+				}
+
+				if($inactiveforums)
+				{
+					$where_sql .= " AND p.fid NOT IN ({$inactiveforums})";
+				}
+
+				// Check group permissions if we can't view threads not started by us
+				$onlyusfids = array();
+				$group_permissions = forum_permissions();
+				foreach($group_permissions as $fid => $forumpermissions)
+				{
+					if(isset($forumpermissions['canonlyviewownthreads']) && $forumpermissions['canonlyviewownthreads'] == 1)
+					{
+						$onlyusfids[] = $fid;
+					}
+				}
+
+				if(!empty($onlyusfids))
+				{
+					$where_sql .= "AND ((t.fid IN(".implode(',', $onlyusfids).") AND t.uid='{$mybb->user['uid']}') OR t.fid NOT IN(".implode(',', $onlyusfids)."))";
+				}
+
+				// Moderators can view unapproved/deleted posts
+				if($mybb->usergroup['issupermod'] != 1)
+				{
+					$unapprove_forums = array();
+					$deleted_forums = array();
+					$visible_sql = ' AND p.visible > 0 AND t.visible > 0';
+					$query = $db->simple_select("moderators", "fid, canviewunapprove, canviewdeleted", "(id='{$mybb->user['uid']}' AND isgroup='0') OR (id='{$mybb->user['usergroup']}' AND isgroup='1')");
+					while($moderator = $db->fetch_array($query))
+					{
+						if($moderator['canviewunapprove'] == 1)
+						{
+							$unapprove_forums[] = $moderator['fid'];
+						}
+
+						if($moderator['canviewdeleted'] == 1)
+						{
+							$deleted_forums[] = $moderator['fid'];
+						}
+					}
+
+					if(!empty($unapprove_forums))
+					{
+						$visible_sql .= " OR (p.visible = 0 AND p.fid IN(".implode(',', $unapprove_forums)."))";
```

Comment:
Doesn't `$visible_sql = ' AND p.visible > 0 AND t.visible > 0';` + this skip unapproved (and soft deleted below) posts?

If I'm not mistaken, `p.visible > 0` is always required by `AND` and the OR applies to `t.visible > 0` and the other 2 conditions.